tpu/hptuning/resnet-hypertune/resnet/imagenet_input.py

```diff
@@ -0,0 +1,211 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Efficient ImageNet input pipeline using tf.data.Dataset."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+import tensorflow as tf
+
+from . import resnet_preprocessing
+
+
+def image_serving_input_fn():
+  """Serving input fn for raw images."""
+
```

Comment:
I think I saw duplicate code above. Serving function etc.

Reply:
`tpu/hptuning/resnet-hypertune` and `tpu/hptuning/resnet-hptuning` are both based on the same official sample.  The goal is to show two different ways of using the hptuning service, while keeping both samples self-contained (as in, clone-runnable).